package.json

```diff
@@ -12,6 +12,7 @@
     "react": "16.8.6",
     "react-dom": "16.8.6",
     "react-redux": "7.0.3",
+    "react-transition-group": "^4.2.1",
```

Comment:
Including @dmose to help review the new package added in the build. Thanks!

Reply:
Yes! It looks like this is officially endorsed by React as its mentioned in the official docs https://reactjs.org/docs/animation.html. 

It is under community support at https://github.com/reactjs/react-transition-group with a BSD-3 license. The addon itself significantly simplifies animations that are based on state/prop changes especially when it comes to mounting/unmounting.

```
Exposes simple components useful for defining entering and exiting transitions. React Transition Group is not an animation library like React-Motion, it does not animate styles by itself. Instead it exposes transition stages, manages classes and group elements and manipulates the DOM in useful ways, making the implementation of actual visual transitions much easier.
```

Reply:
In general, this seems reasonable.  Since it seems to have close ties to the React community, decent test coverage, and what looks to be a compatible license.  @k88hudson, any thoughts?
 
Assuming that Kate's ok with it, there are a few things that would need to happen to land this change:

- [ ] file a license bug as described at https://www.mozilla.org/en-US/MPL/license-policy/ to get formal approval from the licensing team
- [ ] split the vendoring changes into their own commit (and don't squash it, perhaps by putting it into its own PR)
- [ ] for some reason, it's not listed in package-lock.json, which makes me think it's not installed correctly.  Pin it to an exact version ("4.2.1") rather than the less specific "^4.2.1", probably by using the `--save-exact` switch to npm.  Make sure you're using npm 6.9.0 to do the install.
- [ ] post a patch to [`toolkit/content/license.html](https://searchfox.org/mozilla-central/source/toolkit/content/license.html) add the license in appropriately.  This patch could be posted to its own bug, or to the bug for the upcoming export, if there is one.

Does the CSS transition group stuff in this patch, as written, work correctly in the running browser?  Since it doesn't contain a vendored copy for the runtime, I would sort of expect it to fail.  If that's true, it needs to be added to the bin/vendor.js script and that needs to be run once as well.

Reply:
@dmose Looking at activity-stream.bundle.js, react-transition-group is getting added as concatenated module similar to fluent, fluent-react and cached-iterable. How webpack is picking and bundling this module is still not clear to me.